smdebug/tensorflow/keras.py

```diff
@@ -730,6 +730,56 @@ def _prepare_collections_for_tf2(self):
             self._unwrap_model_with_input_output_saver()
         self.prepared_tf2_collections = True
 
+    def _stop_detailed_profiling(self, current_step=None):
+        """Stop detailed profiling if the TF profiler is supported in the current TF version and detailed profiling
+        was already started.
+        """
+        if not is_profiler_supported_for_tf_version() or not self.is_detailed_profiling:
+            return
+
+        if current_step is None:
+            current_step = self.mode_steps[self.mode]
+
+        self.logger.info(f"Disabling TF profiler on step: ={current_step}")
+        stop_tf_profiler(
+            tf_profiler=self.tf_profiler,
+            log_dir=self._log_dir,
+            start_time_us=self.tf_profiler_start_time_in_micros,
+        )
+        self.is_detailed_profiling = False
+
+    def _handle_detailed_profiling(self, current_step):
```

Comment:
nit: try to come up with a better name `_handle_detailed_profiling`

Reply:
Used `start_or_stop_detailed_profiling`